server/service/microsoft_mdm.go

```diff
@@ -1810,10 +1811,34 @@ func (svc *Service) storeWindowsMDMEnrolledDevice(ctx context.Context, userID st
 		}
 	}
 
+	// Get the host in order to get the correct display name and serial number for the activity
```

Comment:
This should be moved inside the `if hostUUID != ""` above, otherwise it will run even if we don't have a uuid.

Reply:
good catch!